### ISSUE TITLE
📎 Port most of the `a11y-jsx` rules under the new `a11y` category rules

### ISSUE THREAD

**Issue by ematipico** [rome/tools#1345]
Recently, we created a new lint category called `a11y`. This new category is meant to be generic and work for JSX and HTML at the same time.

At the moment, we ported only two rules and we should to the same for most of the `a11y-jsx` we currently have

**Comment by mainangethe** [comment]
Hi @ematipico ,
I'd like to take this up. Though I'm not sure what it means when you say you "ported"

Could you give a more details surrounding what I should do to port a rule, so that I can be able to start.

**Comment by ematipico** [comment]
Hi @mainangethe 

You can use this PR as example https://github.com/rome/tools/pull/1346

At the moment we have a bunch of rules that target some `a11y` requirements. These rules are tailored around JSX AST. Now that we have a working HTML AST parser, what we want to do is to make these rules a bit generic.

So we have to take those rules that are under `jsx-a11y` and move them under the category `a11y`. Not only that: we want to add HTML support, which means that one rule will be valid for JSX and HTML at the same time. 

These are the steps we should follow to move an existing JSX rule:

- create a new rule under the category `a11y`; the same of the rule should be the same of the one you plan to work on;
- copy/paste the existing rule inside the new file (and tests too);
- remove from the code base all the references to the old rule, e.g.: `"jsx-a11y/useOldRule"`, `"lint/jsx-a11y/useOldRule"`;
- extend the rule to handle HTML too;
- add new test cases for a HTML file;

**Comment by zant** [comment]
Hi! I'll be working on `noHeaderScope` :)

**Comment by imrishabh18** [comment]
@ematipico Are some more rules left to be ported than I can work on? I am new to this

**Comment by ematipico** [comment]
> @ematipico Are some more rules left to be ported than I can work on? I am new to this

`useHeadingContent` is available! `useAnchorContent` too. They are both simple to implement.

**Comment by ematipico** [comment]
Two rules missing! 🚀 
- `useValidAnchor`
- `noRedundantRoles`

**Comment by harxki** [comment]
I'll be working on useValidAnchor !

**Comment by UsamaHameed** [comment]
I will take on `noRedundantRoles`

**Comment by ematipico** [comment]
@UsamaHameed that's a though one :) If you have any questions, please ask!

**Comment by ematipico** [comment]
I want to thank you everyone that helped to move all the rules! We finished, now all the rules will work on HTML and JSX! 🎉 🎉
